File: src/Tools/Dova.Tools.Java.ClassGenerator/src/main/java/com/github/sejoslaw/dova/ClassDefinitionModel.java
package com.github.sejoslaw.dova;

import java.util.ArrayList;
import java.util.Collection;

public class ClassDefinitionModel {
    public ClassDetailsDefinitionModel ClassDetailsModel = new ClassDetailsDefinitionModel();
    public ClassElementDefinitionModel BaseClassModel = new ClassElementDefinitionModel();
    public Collection<ClassElementDefinitionModel> InterfaceModels = new ArrayList<>();
    public Collection<ClassElementDefinitionModel> ConstructorModels = new ArrayList<>();
    public Collection<ClassElementDefinitionModel> FieldModels = new ArrayList<>();
    public Collection<ClassElementDefinitionModel> MethodModels = new ArrayList<>();
    public Collection<ClassDefinitionModel> InnerClassModels = new ArrayList<>();
}

File: src/Tools/Dova.Tools.JavaClassStructureGenerator/Models/ClassDefinitionModel.cs
namespace Dova.Tools.JavaClassStructureGenerator.Models;

internal class ClassDefinitionModel
{
    public ClassDetailsDefinitionModel ClassDetailsModel { get; set; }
    public ClassElementDefinitionModel BaseClassModel { get; set; }
    public IReadOnlyList<ClassElementDefinitionModel> InterfaceModels { get; set; } = new List<ClassElementDefinitionModel>();
    public IReadOnlyList<ClassElementDefinitionModel> ConstructorModels { get; set; } = new List<ClassElementDefinitionModel>();
    public IReadOnlyList<ClassElementDefinitionModel> FieldModels { get; set; } = new List<ClassElementDefinitionModel>();
    public IReadOnlyList<ClassElementDefinitionModel> MethodModels { get; set; } = new List<ClassElementDefinitionModel>();
    public IReadOnlyList<ClassDefinitionModel> InnerClassModels { get; set; } = new List<ClassDefinitionModel>();
}